lib/mocha.js

```diff
@@ -482,21 +509,32 @@ Mocha.prototype.run = function(fn) {
   if (options.globals) {
     runner.globals(options.globals);
   }
+  // Use only the first reporter for growl, since we don't want to
+  // send several notifications for the same test suite
   if (options.growl) {
-    this._growl(runner, reporter);
+    this._growl(runner, reporters[0]);
   }
   if (options.useColors !== undefined) {
     exports.reporters.Base.useColors = options.useColors;
   }
   exports.reporters.Base.inlineDiffs = options.useInlineDiffs;
 
-  function done(failures) {
-    if (reporter.done) {
-      reporter.done(failures, fn);
-    } else {
-      fn && fn(failures);
+  function runnerDone(failures) {
+    function reporterDone(failures) {
+      if (--remain === 0) {
+        fn && fn(failures);
```

Comment:
I think it's safer, since `reporterDone` might be called as a callback by `reporter.done`. I'll use `reporerFailures` as a name to avoid shadowing the variable which might cause some confusion.